app/src/test/java/ch/epfl/sdp/drone3d/model/util/GeometryUtilsTest.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.model.util
+
+import com.mapbox.mapboxsdk.geometry.LatLng
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class GeometryUtilsTest{
+
+    @Test
+    fun getFourthParallelogramPointOnSquareIsCorrect(){
+
+        val origin = LatLng(0.0,0.0)
+        val topAdjacent =  LatLng(1.0,0.0)
+        val rightAdjacent =  LatLng(0.0,1.0)
+
+        val expected= LatLng(1.0,1.0)
+        val actual = GeometryUtils.getFourthParallelogramVertex(origin,topAdjacent,rightAdjacent)
+
+        assertEquals(expected.latitude,actual.latitude,0.0001)
+        assertEquals(expected.longitude,actual.longitude,0.0001)
+    }
+
+    @Test
+    fun getFourthParallelogramPointOnBoundarySquareIsCorrect(){
+
+        val origin =  LatLng(0.0,-179.0)
+        val rightAdjacent = LatLng(0.0,179.0)
+        val topAdjacent =  LatLng(1.0,-179.0)
+
+        val expected = LatLng(1.0,-177.0)
```

Comment:
The expected value is not correct. It should be LatLng(1.0, 179). Since the test is passing, it means that we probably have an error in the SphereToPlaneProjector. I think that the error comes from the fact that we just check if the longitude is bigger than the origin to choose between +x and - x  and that it is a boarder case that we didn't take in account. Shouldn't be too hard to correct though. We can speak about it and correct it together.